Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -1108,17 +1108,12 @@ void FunctionDefinitionWriter::visit(const Type* type, AST::Expression& expressi
         return;
     }
 
-    switch (expression.kind()) {
```

Comment:
Alternatively we could keep this a switch and use `uncheckedDowncast<>`. I think that's the way to go in some other WebGPU code so let me know if you'd prefer that here as well.